plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py

```diff
@@ -4,12 +4,17 @@
 import os
 import os.path
 import logging
+import sys
 from typing import List, Optional, Iterator, Tuple, Any, Dict, Deque
 
 from dataclasses import dataclass
-from cached_property import cached_property
 from functools import partial
 
+if sys.version_info >= (3, 8):
```

Comment:
Nitpick: Do you think it'd be worth creating a module to encapsulate this code, e.g. `src/sqlfluff/core/cached_property.py`? That would save us having to repeat the version checking and conditional import logic in each file (current and future) that uses cached properties.

Reply:
yeah sure thing!